python_modules/dagster/dagster_tests/definitions_tests/auto_materialize_tests/scenarios/exotic_partition_mapping_scenarios.py

```diff
@@ -310,6 +326,93 @@
         expected_run_requests=[run_request(asset_keys=["asset1"], partition_key="2020-01-02")],
         current_time=create_pendulum_time(year=2020, month=1, day=3, hour=4),
     ),
+    "self_dependency_prior_partition_rematerialized": AssetReconciliationScenario(
+        assets=one_asset_self_dependency,
+        # this is rematerialized
+        unevaluated_runs=[single_asset_run(asset_key="asset1", partition_key="2020-01-01")],
+        cursor_from=AssetReconciliationScenario(
+            assets=one_asset_self_dependency,
+            unevaluated_runs=[
+                single_asset_run(asset_key="asset1", partition_key="2020-01-01"),
+                single_asset_run(asset_key="asset1", partition_key="2020-01-02"),
+                single_asset_run(asset_key="asset1", partition_key="2020-01-03"),
+            ],
+            expected_run_requests=[],
+            current_time=create_pendulum_time(year=2020, month=1, day=4, hour=4),
+        ),
+        # should not rematerialize partition because of this update
+        expected_run_requests=[],
+        current_time=create_pendulum_time(year=2020, month=1, day=4, hour=5),
+    ),
+    "self_dependency_upstream_prior_partition_rematerialized": AssetReconciliationScenario(
+        assets=three_assets_self_dependency,
+        unevaluated_runs=[single_asset_run(asset_key="asset1", partition_key="2020-01-01")],
+        cursor_from=AssetReconciliationScenario(
+            assets=three_assets_self_dependency,
+            unevaluated_runs=[
+                run(["asset1", "asset2", "asset3"], partition_key="2020-01-01"),
+                run(["asset1", "asset2", "asset3"], partition_key="2020-01-02"),
+                run(["asset1", "asset2", "asset3"], partition_key="2020-01-03"),
+            ],
+            expected_run_requests=[],
+            current_time=create_pendulum_time(year=2020, month=1, day=4, hour=4),
+        ),
+        # should respond to an upstream asset being rematerialized
+        expected_run_requests=[run_request(["asset2", "asset3"], partition_key="2020-01-01")],
+        current_time=create_pendulum_time(year=2020, month=1, day=4, hour=4),
+    ),
+    "self_dependency_upstream_prior_partition_rematerialized_with_not_all_parents_updated": AssetReconciliationScenario(
+        assets=with_auto_materialize_policy(
+            three_assets_self_dependency,
+            AutoMaterializePolicy.eager().with_rules(
+                AutoMaterializeRule.skip_on_not_all_parents_updated()
+            ),
+        ),
+        unevaluated_runs=[single_asset_run(asset_key="asset1", partition_key="2020-01-01")],
```

Comment:
nit: might be more accurate to rematerialize 2020-01-02? since 2020-01-01 in asset2 doesn't rely on any prior partitions in asset2

Reply:
hm yeah doesn't hurt to choose a trickier example here